tsdb/db.go

```diff
@@ -45,19 +48,27 @@ import (
 
 // Default duration of a block in milliseconds - 2h.
 const (
-	DefaultBlockDuration = int64(2 * 60 * 60 * 1000)
+	DefaultBlockDuration = int64(2 * 60 * 60 * 1000) // 2h in miliseconds.
+)
+
+var (
+	// ErrNotReady is returned if the underlying storage is not ready yet.
+	ErrNotReady = errors.New("TSDB not ready")
 )
 
 // DefaultOptions used for the DB. They are sane for setups using
 // millisecond precision timestamps.
-var DefaultOptions = &Options{
-	WALSegmentSize:         wal.DefaultSegmentSize,
-	RetentionDuration:      15 * 24 * 60 * 60 * 1000, // 15 days in milliseconds
-	BlockRanges:            ExponentialBlockRanges(DefaultBlockDuration, 3, 5),
-	NoLockfile:             false,
-	AllowOverlappingBlocks: false,
-	WALCompression:         false,
-	StripeSize:             DefaultStripeSize,
+func DefaultOptions() *Options {
+	return &Options{
+		WALSegmentSize:         wal.DefaultSegmentSize,
+		RetentionDuration:      15 * 24 * model.Duration(time.Hour),
+		MinBlockDuration:       model.Duration(time.Duration(DefaultBlockDuration) * time.Millisecond),
```

Comment:
model has no place in tsdb, among other things the tsdb does not know anything about the units it is being used with.

Reply:
I see.... I can make that happen, but why tsdb cannot import `model`?

Reply:
And thanks to that have much nicer UX? (units are embedded in type)

Reply:
I guess that's the reason: https://github.com/prometheus/prometheus/pull/6777#discussion_r377715609